### DISCUSSION TITLE
How can I append response header on next.js ?

### DISCUSSION THREAD

**Discussion by cmlbrnc** [vercel/next.js#14092] [Help]
It will be very cool if I am setting the response header in next.config.js.

I should add  ``` headers['strict-transport-security'] = [{key: 'Strict-Transport-Security', value: 'max-age=63072000; includeSubdomains; preload'}]; ``` header to  https request for seo purpose.  I cant integrate express.js with the next.js because of using aws-serverless-next.js framework.  I also cant edit my cloudEdge@ lambda function. 
So any advice to solve this proplem for now will be helpful. Thank you in advance.

**Comment by srikanth912** [comment]
If you are using _app.js, then maybe you can do that in getInitialProps using res.writeHead. 

We use the below snippet to indicate google page has moved to a new location permanently. (a stripped bunch of logic). We also same to set some cookies too using the same method. 

```
static async getInitialProps({ Component, ctx: {res} }) {
  res.writeHead(301, { Location: `https://${store.domain}` });
}
```

In your case maybe you need to use below (or something similar) as per [this](https://stackoverflow.com/questions/37014433/client-browser-how-to-set-standard-security-headers-https-server-on-wss-connec):

`res.writeHead(200, {"Strict-Transport-Security": "max-age=63072000", includeSubdomains: "preload"})
`

**Reply by cmlbrnc** [reply]
@srikanth912  Thank you. I tried it. I am getting error because  it is already sent. 
Error [ERR_HTTP_HEADERS_SENT]: Cannot set headers after they are sent to the client

**Comment by Timer** [comment]
You can create a `next.config.js` with the following:
```js
// next.config.js
module.exports = {
  experimental: {
    headers() {
      return [
        {
          source: "/:all*",
          headers: [
            {
              key: "Strict-Transport-Security",
              value: "max-age=63072000; includeSubdomains; preload",
            },
          ],
        },
      ];
    },
  },
};

```

**Reply by cmlbrnc** [reply]
Thank you for answer. It doesnt work for me. I am not sure but maybe  this is manipulating by the serverless-next.js and doesnt return the response headers I want. So Maybe I should move this topic under serverless-next.js.

**Reply by Timer** [reply]
Indeed, this would be a bug with aws-serverless-next.js then.

**Reply by Timer** [reply]
Deploying on Vercel (also a serverless platform) does work, so I suggest you give that a try (we also auto configure this header)!

**Reply by cmlbrnc** [reply]
My local dev server is not append the header as well. There must be another problem.

**Comment by cmlbrnc** [comment]
please see the code below. It doesnt work as well. I need a bit more serverless.yml knowledge. Can you guys help me  ? Thank you.
```
nextApp:
  component: "serverless-next.js"
  inputs:
    cloudfront:
    
      /*:
       response:
            headers:
  
               Strict-Transport-Security: "'max-age=31536000; includeSubDomains; preload'"
        ```
